From ee625b2d8acad2ba81693107740a12d8a03863c0 Mon Sep 17 00:00:00 2001
From: Aleix Pol <aleixpol@kde.org>
Date: Mon, 28 Mar 2022 03:40:12 +0200
Subject: [PATCH] Fix unlocking wayland sessions

We were never issuing a repaint which confused thoroughly the
compositor.

BUG: 447705
---
 src/wayland_server.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/wayland_server.cpp b/src/wayland_server.cpp
index eb3600f824..9e60787465 100644
--- a/src/wayland_server.cpp
+++ b/src/wayland_server.cpp
@@ -15,6 +15,7 @@
 #include "inputpanelv1integration.h"
 #include "keyboard_input.h"
 #include "screens.h"
+#include "scene.h"
 #include "layershellv1integration.h"
 #include "main.h"
 #include "xdgshellintegration.h"
@@ -610,6 +611,7 @@ void WaylandServer::initScreenLocker()
                 connect(seat, &KWaylandServer::SeatInterface::timestampChanged,
                         screenLockerApp, &ScreenLocker::KSldApp::userActivity);
             }
+            Compositor::self()->scene()->addRepaintFull();
         }
     );
 
@@ -627,6 +629,7 @@ void WaylandServer::initScreenLocker()
                            screenLockerApp, &ScreenLocker::KSldApp::userActivity);
             }
             ScreenLocker::KSldApp::self()->setWaylandFd(-1);
+            Compositor::self()->scene()->addRepaintFull();
         }
     );
 
-- 
GitLab

